ui/src/main/java/org/pentaho/di/ui/job/entry/JobEntryDialog.java

```diff
@@ -303,7 +309,19 @@ String showDbDialogUnlessCancelledOrValid( DatabaseMeta changing, DatabaseMeta o
           // OK was pressed and input is valid
           repeat = false;
         } else {
-          showDbExistsDialog( changing );
+          try {
+            DatabaseManagementInterface dbMgr =
+              spoonSupplier.get().getBowl().getManager( DatabaseManagementInterface.class );
+            if ( dbMgr.getDatabases().stream().anyMatch( db -> db.getName().equals( changing.getName() ) ) ) {
```

Comment:
getDatabase( name ) will again be more efficient. 

The call up on line 306 is sketchy as well. The name needs to be checked against others only at its existing level, even though the dialog is showing a flattened list.

Reply:
I've refactored the collision detection logic.  Had to revert back to getDatabases so that I could do case insensitive comparisons.